code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -51,91 +60,153 @@
 	add_fingerprint(user)
 	return
 
+/obj/item/clothing/suit/armor/reactive/hit_reaction(owner, hitby, attack_text, final_block_chance, damage, attack_type)
+	if(!active || !prob(hit_reaction_chance))
+		return FALSE
+	if(world.time < reactivearmor_cooldown)
+		cooldown_activation(owner)
+		return FALSE
+	if(bad_effect)
+		. = emp_activation(owner, hitby, attack_text, final_block_chance, damage, attack_type)
+	else
+		. = reactive_activation(owner, hitby, attack_text, final_block_chance, damage, attack_type)
+
+/**
+ * A proc for doing cooldown effects (like the sparks on the tesla armor, or the semi-stealth on stealth armor)
+ * Called from the suit activating whilst on cooldown.
+ * You should be calling ..()
+ */
+/obj/item/clothing/suit/armor/reactive/proc/cooldown_activation(mob/living/carbon/human/owner)
+	owner.visible_message(cooldown_message)
+
+/**
+ * A proc for doing reactive armor effects.
+ * Called from the suit activating while off cooldown, with no emp.
+ * Returning TRUE will block the attack that triggered this
+ */
+/obj/item/clothing/suit/armor/reactive/proc/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive armor doesn't do much! No surprises here.</span>")
+	return TRUE
+
+/**
+ * A proc for doing owner unfriendly reactive armor effects.
+ * Called from the suit activating while off cooldown, while the armor is still suffering from the effect of an EMP.
+ * Returning TRUE will block the attack that triggered this
+ */
+/obj/item/clothing/suit/armor/reactive/proc/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive armor doesn't do much, despite being emp'd! Besides giving off a special message, of course.</span>")
+	return TRUE
+
 /obj/item/clothing/suit/armor/reactive/emp_act(severity)
 	. = ..()
-	if(. & EMP_PROTECT_SELF)
+	if(. & EMP_PROTECT_SELF || bad_effect || !active) //didn't get hit or already emp'd, or off
 		return
-	active = FALSE
-	icon_state = "reactiveoff"
-	inhand_icon_state = "reactiveoff"
-	reactivearmor_cooldown = world.time + 200
+	visible_message(emp_message)
+	bad_effect = TRUE
+	addtimer(VARSET_CALLBACK(src, bad_effect, FALSE), 30 SECONDS)
```

Comment:
note that, due to the way that you're handling this, if someone EMPs your reactive armor, then EMPs it again 29 seconds later, the emp effect will end 30 seconds after the first EMP/1 second after the second EMP. in other words, you can't reset the duration of the EMP effect while it's active by just emping the armor again before the EMP effect runs out.